frontend/server/src/Controllers/Problem.php

```diff
@@ -3816,13 +3816,51 @@ public static function getProblemEditDetailsForSmarty(
             $r,
             /*$isRequired=*/ false
         );
-
+        $validatorsTypes = [
+            \OmegaUp\ProblemParams::VALIDATOR_TOKEN_CASELESS => \OmegaUp\Translations::getInstance()->get(
+                'problemEditFormTokenCaseless'
+            ),
+            \OmegaUp\ProblemParams::VALIDATOR_TOKEN_NUMERIC => \OmegaUp\Translations::getInstance()->get(
+                'problemEditFormNumericTokensWithTolerance'
+            ),
+            \OmegaUp\ProblemParams::VALIDATOR_TOKEN => \OmegaUp\Translations::getInstance()->get(
+                'problemEditFormTokenByToken'
+            ),
+            \OmegaUp\ProblemParams::VALIDATOR_LITERAL => \OmegaUp\Translations::getInstance()->get(
+                'problemEditFormLiteral'
+            ),
+            \OmegaUp\ProblemParams::VALIDATOR_CUSTOM => \OmegaUp\Translations::getInstance()->get(
+                'problemEditFormCustom'
+            )
+        ];
+        $sortedLanguages = \OmegaUp\Controllers\Run::DEFAULT_LANGUAGES;
+        sort($sortedLanguages);
+        $validLanguages = [ join(
+            ',',
+            $sortedLanguages
+        ) => 'C, C++, C++11, C#, Haskell, Java, Pascal, Python, Ruby, Lua',
+        'kj,kp' => 'Karel',
+        'cat' => \OmegaUp\Translations::getInstance()->get(
+            'wordsJustOutput'
+        ),
+        '' => \OmegaUp\Translations::getInstance()->get(
+            'wordsNoSubmissions'
+        )];
         if (!isset($r['request'])) {
             return [
                 'IS_UPDATE' => true,
                 'LOAD_MATHJAX' => true,
                 'LOAD_PAGEDOWN' => true,
                 'STATUS_SUCCESS' => '',
+                'LANGUAGES' => join(
+                    ',',
+                    \OmegaUp\Controllers\Run::DEFAULT_LANGUAGES
+                ),
+                'payload' => [
+                    'languages' => \OmegaUp\Controllers\Run::DEFAULT_LANGUAGES,
+                    'validLanguages' => $validLanguages,
+                    'validatorsTypes' => $validatorsTypes,
+                    ]
```

Comment:
nit:

```suggestion
                ],
```

los corchetes / paréntesis se cierran en la primer columna no-vacía donde se abren.

Reply:
listo